langchain/document_loaders/generic.py

```diff
@@ -0,0 +1,131 @@
+        from langchain.document_loaders.blob_loaders import FileSystemBlobLoader
+
+        loader = GenericLoader.from_filesystem(
+            path="path/to/directory",
+            glob="**/[!.]*",
+            suffixes=[".pdf"],
+            show_progress=True,
+        )
+
+        docs = loader.lazy_load()
+        next(docs)
+
+        Example instantiations to change which files are loaded:
+
+        ... code-block:: python
+
+            # Recursively load all text files in a directory.
+            loader = GenericLoader.from_filesystem("/path/to/dir", glob="**/*.txt")
+
+            # Recursively load all non-hidden files in a directory.
+            loader = GenericLoader.from_filesystem("/path/to/dir", glob="**/[!.]*")
+
+            # Load all files in a directory without recursion.
+            loader = GenericLoader.from_filesystem("/path/to/dir", glob="*")
+
+        Example instantiations to change which parser is used:
+
+        ... code-block:: python
+
+            from langchain.document_loaders.parsers.pdf import PyPDFParser
+
+            # Recursively load all text files in a directory.
+            loader = GenericLoader.from_filesystem(
+                "/path/to/dir",
+                glob="**/*.pdf",
+                parser=PyPDFParser()
+            )
+    """
+
+    def __init__(
+        self,
+        blob_loader: BlobLoader,
+        blob_parser: BaseBlobParser,
+    ) -> None:
+        """A generic document loader.
+
+        Args:
+            blob_loader: A blob loader which knows how to yield blobs
+            blob_parser: A blob parser which knows how to parse blobs into documents
+        """
+        self.blob_loader = blob_loader
+        self.blob_parser = blob_parser
+
+    def lazy_load(
+        self,
+    ) -> Iterator[Document]:
+        """Load documents lazily. Use this when working at a large scale."""
+        for blob in self.blob_loader.yield_blobs():
+            yield from self.blob_parser.lazy_parse(blob)
+
+    def load(self) -> List[Document]:
+        """Load all documents."""
+        return list(self.lazy_load())
+
+    def load_and_split(
+        self, text_splitter: Optional[TextSplitter] = None
+    ) -> List[Document]:
+        """Load all documents and split them into sentences."""
+        raise NotImplementedError(
+            "Loading and splitting is not yet implemented for generic loaders. "
+            "When they will be implemented they will be added via the initializer. "
+            "This method should not be used going forward."
+        )
+
+    @classmethod
+    def from_filesystem(
+        cls,
+        path: _PathLike,
+        *,
+        glob: str = "**/[!.]*",
+        suffixes: Optional[Sequence[str]] = None,
+        show_progress: bool = False,
+        parser: Union[str, BaseBlobParser] = "default",
+    ) -> GenericLoader:
+        """Create a generic document loader using a filesystem blob loader.
+
+        Args:
```

Comment:
should we add types to these, and match signature order

Reply:
order yes, i suggest not to add types since the type is already provided as part of the function signature